src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +395,467 @@ else if (!period.isEmpty()) {
+        // This will never return NPE as htmlNode will at have least one child
+        // This child will have at least 3 children
+        // The first child is of type HTML_ELEMENT_START which always have at least 2 elements.
+        return htmlNode.getChildren()[0].getChildren()[0].getChildren()[1].getText();
+    }
+
+    /**
+     * Get text inside html elements.
+     *
+     * @param htmlNode html element node.
+     * @return text inside html element.
+     */
+    private static String getStringInsideHtmlElement(DetailNode htmlNode) {
+        final StringBuilder htmlString = new StringBuilder(250);
+        // This will never return NPE as htmlNode will at have least one child
+        // This child will have at least 3 children
+        for (DetailNode child: htmlNode.getChildren()) {
+            if (child.getType() == JavadocTokenTypes.TEXT) {
+                htmlString.append(child.getText());
+            }
+        }
+        return htmlString.toString();
+    }
+
+    /**
+     * Gets summary sentence for html inline type summary.
+     * Html type summary sentence are wrapped within paragraph tag.
+     *
+     * @param ast Javadoc root node.
+     * @return first sentence.
+     */
+    private static String getInlineHtmlSentence(DetailNode ast) {
+        final StringBuilder result = new StringBuilder(256);
+        final DetailNode node = getInlineTagNodeInsideHtmlTagsOnly(ast);
+        result.append(getFirstSentenceFromInlineTagNode(
+                Objects.requireNonNull(node)));
+        return result.toString();
+    }
+
+    /**
+     * Finds and returns description node.
+     *
+     * @param ast Javadoc root node.
+     * @return Description DetailNode.
+     */
+    private static DetailNode getDescriptionNode(DetailNode ast) {
+        DetailNode node = null;
+        for (DetailNode child : ast.getChildren()) {
+            if (child.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                node = JavadocUtil.findFirstToken(child, JavadocTokenTypes.DESCRIPTION);
+            }
+        }
+        return node;
+    }
+
+    /**
+     * Tests if first sentence contains forbidden summary fragment.
+     *
+     * @param firstSentence String with first sentence.
+     * @return true, if first sentence contains forbidden summary fragment.
+     */
+    private boolean containsForbiddenFragment(String firstSentence) {
+        final String javadocText = JAVADOC_MULTILINE_TO_SINGLELINE_PATTERN
+                .matcher(firstSentence).replaceAll(" ").trim();
+        return forbiddenSummaryFragments.matcher(trimExcessWhitespaces(javadocText)).find();
+    }
+
+    /**
+     * Trims the given {@code text} of duplicate whitespaces.
+     *
+     * @param text The text to transform.
+     * @return The finalized form of the text.
+     */
+    private static String trimExcessWhitespaces(String text) {
+        final StringBuilder result = new StringBuilder(100);
```

Comment:
make 256, to show that you do not actually know, and just want to have some big default.

Reply:
Done